devicetypes/smartthings/smartpower-outlet.src/smartpower-outlet.groovy

```diff
@@ -45,80 +44,62 @@ metadata {
+			])
 		}
 	}
 
 	// UI tile definitions
 	tiles(scale: 2) {
-		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
-			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
 				attributeState "on", label: 'On', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#79b821", nextState: "turningOff"
 				attributeState "off", label: 'Off', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff", nextState: "turningOn"
 				attributeState "turningOn", label: 'Turning On', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#79b821", nextState: "turningOff"
 				attributeState "turningOff", label: 'Turning Off', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff", nextState: "turningOn"
 			}
-			tileAttribute ("power", key: "SECONDARY_CONTROL") {
-				attributeState "power", label:'${currentValue} W'
+			tileAttribute("power", key: "SECONDARY_CONTROL") {
+				attributeState "power", label: '${currentValue} W'
 			}
 		}
 
 		standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
-			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
 		}
 
 		main "switch"
-		details(["switch","refresh"])
+		details(["switch", "refresh"])
 	}
 }
 
 // Parse incoming device messages to generate events
 def parse(String description) {
 	log.debug "description is $description"
 
-	def finalResult = zigbee.getKnownDescription(description)
-	def event = [:]
-
-	//TODO: Remove this after getKnownDescription can parse it automatically
-	if (!finalResult && description!="updated")
-		finalResult = getPowerDescription(zigbee.parseDescriptionAsMap(description))
+	def event = zigbee.getEvent(description)
 
-	if (finalResult) {
-		log.info "final result = $finalResult"
-		if (finalResult.type == "update") {
-			log.info "$device updates: ${finalResult.value}"
-			event = null
-		}
-		else if (finalResult.type == "power") {
-			def powerValue = (finalResult.value as Integer)/10
-			event = createEvent(name: "power", value: powerValue, descriptionText: '{{ device.displayName }} power is {{ value }} Watts', translatable: true)
-			/*
-				Dividing by 10 as the Divisor is 10000 and unit is kW for the device. AttrId: 0302 and 0300. Simplifying to 10
-				power level is an integer. The exact power level with correct units needs to be handled in the device type
-				to account for the different Divisor value (AttrId: 0302) and POWER Unit (AttrId: 0300). CLUSTER for simple metering is 0702
-			*/
+	if (event) {
+		if (event.name == "power") {
+			event.value = event.value / 10
```

Comment:
You could use the same comment here that you have in smartpower-dimming-outlet.groovy